api-server/config.py

```diff
@@ -1,8 +1,34 @@
 import os
 from dotenv import load_dotenv
+import argparse
 
 load_dotenv()
 
 GITHUB_ACCESS_TOKEN = os.environ.get("GITHUB_ACCESS_TOKEN")
 DB_URI = os.environ.get("DB_URI")
 TEST_DB_URI = os.environ.get("TEST_DB_URI")
+GITHUB_CLIENT_ID = os.environ.get("GITHUB_CLIENT_ID")
+GITHUB_CLIENT_SECRET = os.environ.get("GITHUB_CLIENT_SECRET")
+GITHUB_ACCESS_TOKEN_ENDPOINT = os.environ.get("GITHUB_ACCESS_TOKEN_ENDPOINT")
+GITHUB_LOGIN_ENDPOINT = os.environ.get("GITHUB_LOGIN_ENDPOINT")
+AUTH_SECRET = os.environ.get("AUTH_SECRET")
+
+"""
+Added this bit here to make local testing easy. Run `python main.py -da -u <YOUR_GH_USERNAME>`
+
+Example: `python main.py -da -u ramko9999` to run app without auth as ramko9999
+"""
+parser = argparse.ArgumentParser()
+
+parser.add_argument("-u", "--username", required=False)
+parser.add_argument("-da", "--disable-auth", action="store_true", required=False)
+
+args = parser.parse_args()
+
+DISABLE_AUTH = args.disable_auth
+USERNAME = args.username
+
+if DISABLE_AUTH and not USERNAME:
```

Comment:
Why is `DISABLE_AUTH` coupled with `USERNAME`? Is it not possible to run `python main.py -da`? What happens if I run `python main,py -u <USERNAME>`?

Reply:
Even with these flags, One issue that remains with manual testing is that we cannot impersonate multiple users on different tabs. Upon invoking the server with `-da`, I can bypass authentication for only 1 username. 

**Alternate Approach**
1. Keep the changes for parsing the `disable_auth` flag
2. Create another endpoint (perhaps within the `User` router) hoisted at `/user/impersonate/:IMPERSONATED_USERNAME`. The handler for this can perform the following steps: create a token from the `IMPERSONATED_USERNAME` path parameter and redirect back to the client with the token set as an HTTP-only cookie (exactly like you do [here](https://github.com/rohanshiva/gitgame/pull/60/files#diff-5a2f400b16dae954159ee1d391b9c473bd95d53a36e154b34c4fe49727ed9d96R68-R69)). Only allow for the token generation/redirect to occur if the `disable_auth` flag is turned on.

Now, I can impersonate any user during manual testing and can have different tabs corresponding to different users.